src/utils/internalLinks.js

```diff
@@ -61,9 +62,23 @@ export const getLinkType = (url: string, realm: string): LinkType => {
   return 'home';
 };
 
-/** PRIVATE -- exported only for tests. */
-export const transformToEncodedURI = (string: string): string =>
-  string.replace(/\.\d/g, (match, p1, p2, offset, str) => `%${match[1]}`);
+/** Decode MediaWiki-style dot-encoding. */
```

Comment:
Hmm interesting -- I don't recognize the MediaWiki connection. Is there a useful link?

Some googling finds references to a few kinds of URL encoding in MediaWiki, but they seem to all use percents:
https://en.wikipedia.org/wiki/Help:URL#Fixing_links_with_unsupported_characters
https://www.mediawiki.org/wiki/Manual:PAGENAMEE_encoding

Reply:
_b)_ [It's gone now.](https://www.mediawiki.org/wiki/Manual:%24wgExperimentalHtmlIds)

_a)_ It wasn't used for all URL encoding – just for encoding the names of [section headers](https://web.archive.org/web/20071203202308/http://en.wikipedia.org/wiki/Joseph-Louis_Lagrange#M.C3.A9canique_analytique) and the like. HTML4 (!) required that all `name` attributes consist of [a restricted set of characters](https://www.w3.org/TR/html4/types.html#type-name) which excluded `%` but included `.`.

> Is there a useful link?

The best I can find is https://stackoverflow.com/questions/36880245. I think omitting the word "MediaWiki" and just linking to `hash_util.js` (as suggested below) would be more useful.

Reply:
Ah, that looks familiar now!

Chasing links from there, this might be the best description:
https://phabricator.wikimedia.org/T152540
(That was the issue for removing it.)

That solution SGTM.